pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -374,9 +413,26 @@ func GrowCluster(l *logger.Logger, c *Cluster, numNodes int) error {
 				c.Name, gce.ProviderName)
 		}
 	}
-	return vm.ForProvider(provider, func(p vm.Provider) error {
-		return p.Grow(l, c.VMs, c.Name, names)
+
+	var vmsLock syncutil.Mutex
+	err := vm.ForProvider(provider, func(p vm.Provider) error {
+		addedVms, err := p.Grow(l, c.VMs, c.Name, names)
+		if err != nil {
+			return err
+		}
+
+		// Update the list of VMs in the cluster.
+		vmsLock.Lock()
+		defer vmsLock.Unlock()
+		c.VMs = append(c.VMs, addedVms...)
+
+		return nil
 	})
+	if err != nil {
```

Comment:
Nit: redundant check, `return err` suffices.

Reply:
Good catch! Not true anymore after addressing next comment.